### ISSUE TITLE
508-defect-1 ❗ Launchblocker [SEMANTIC MARKUP]: Form inputs MUST be paired with labels

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#9865]
# [508-defect-1 :exclamation: Launchblocker](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-1)

**Feedback framework**
- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

Form elements must have labels. On 2 of 3 Select your supplies - Select the hearing aids that need batteries, the checkbox input id attributes must be paired with label for attributes.

### Why it Matters
Effective form labels are required to make forms accessible. The purpose of form elements such as checkboxes, radio buttons, input fields, etcetera, is often apparent to sighted users, even if the form element is not programmatically labeled. Screen readers users require useful form labels to identify form fields. Adding a label to all form elements eliminates ambiguity and contributes to a more accessible product.

When labels for form elements are absent, screen reader users do not know the input data expectations. Screen readers cannot programmatically determine information about input objects without an established label relationship (or redundant text serving as a label).

The absence of labels prevent fields from receiving focus when read by screen readers, and users with impaired motor control do not get the benefit of a larger clickable area for the control since clicking the label activates the control.

## Point of Contact

**VFS Point of Contact:** Jennifer

## Acceptance Criteria

* As a screen reader user, I want to know the input data expectations.
* As a screen reader user, I want the input to receive focus and hear the corresponding label.
* As a user with mobility considerations, I want access to the full clickable area for the form control.

## Environment

* Operating System: all
* Browser: all
* Server destination: staging

## Steps to Recreate

1. Enter https://staging.va.gov/hearing-aid-batteries-and-accessories/batteries  in browser, logged in as user 1
2. Navigate to 2 of 3 Select your supplies
3. Run axe browser scan
4. Verify that "Form elements must have labels" is a result associated with `<input name="1" class="vads-u-margin-left--0 vads-u-max-width--293" type="checkbox">`

## Possible Fixes (optional)

Programmatically associate labels with all form controls. The recommended method for most circumstances is to use the label element and an explicit association using the for and id attributes:

`<label for="firstname">First name:</label> <input type="text" id="firstname">`

The label can also be implicit by wrapping the <label> element around the input:

`<label>First name: <input type="text"></label>`

The label has a `for="1"`, but the input does not have a matching id attribute and value.

### Current code

```diff
<div class="vads-u-max-width--293">
!  <input name="1" class="vads-u-margin-left--0 vads-u-max-width--293" type="checkbox">
!  <label class="usa-button vads-u-font-weight--bold vads-u-border--2px vads-u-border-color--primary vads-u-background-color--white vads-u-color--primary" for="1">
    Order batteries for this device
  </label>
</div>
```
### Recommended code

```diff
<div class="vads-u-max-width--293">
+  <input id="1" name="1" class="vads-u-margin-left--0 vads-u-max-width--293" type="checkbox">
+  <label class="usa-button vads-u-font-weight--bold vads-u-border--2px vads-u-border-color--primary vads-u-background-color--white vads-u-color--primary" for="1">
    Order batteries for this device
  </label>
</div>
```

## WCAG or Vendor Guidance (optional)

* [W3C - associating labels with form inputs](https://www.w3.org/TR/WCAG20-TECHS/H44.html)

* [Form `<input>` elements must have labels](https://dequeuniversity.com/rules/axe/3.5/label)

## Screenshots or Trace Logs

<img width="1288" alt="Screen Shot 2020-06-05 at 9 02 04 AM" src="https://user-images.githubusercontent.com/57469/83879368-83723400-a70b-11ea-8b5b-569e311bc363.png">

**Comment by bradl3yC** [comment]
@jenstrickland can you validate in staging?

**Comment by jenstrickland** [comment]
Validated. 🎉
